addons/website_forum/controllers/website_forum.py

```diff
@@ -38,7 +40,11 @@ def _prepare_user_values(self, **kwargs):
         if kwargs.get('forum'):
             values['forum'] = kwargs.get('forum')
         elif kwargs.get('forum_id'):
-            values['forum'] = request.env['forum.forum'].browse(int(kwargs.pop('forum_id')))
+            values['forum'] = Forum.browse(int(kwargs.pop('forum_id')))
+        forum = values.get('forum')
+        if isinstance(forum, Forum.__class__):
```

Comment:
```suggestion
        if isinstance(forum := values.get('forum'), Forum.__class__):
```

Reply:
Thanks for the tricks. But having added an elif based on forum (to match == "all"), I think it is better now to keep the variable before the if that is now a "switch case" based on the forum variable. What do you think ?